### ISSUE TITLE
Group-2: Implement Stacking Ensemble with Multioutput for Binary Classification and Regression (or Multiclass)

### ISSUE THREAD

**Issue by project-delphi** [ccm-bioinfo/Camda24_resistance#12]
**Objective**:
Build a stacking ensemble for a multioutput problem. One output is binary classification, and the other is either regression or multiclass classification (16 classes). Use a variety of base models and a bagging neural network as the final stacking model.

#### Tasks to Complete

1. **Define Base Models for Stacking**

   **Base Models:**
   - [ ] **SVM for Classification**: Use `SVC` for binary classification tasks.
   - [ ] **SVM for Regression**: Use `SVR` for the regression or multiclass task.
   - [ ] **Boosting Decision Trees**: Use `GradientBoostingClassifier` and `GradientBoostingRegressor`.
   - [ ] **Random Forest**: Implement `RandomForestClassifier` and `RandomForestRegressor`.
   - [ ] **L2 Regression**: Use `RidgeClassifier` (for classification) and `Ridge` (for regression).

   - [ ] Train each base model and verify their outputs for multioutput tasks.
   
2. **Stacking Model Using Bagging Neural Network**
   - [ ] Implement a simple neural network using `MLPClassifier` or `MLPRegressor` as the final meta-model.
   - [ ] Use a bagging approach (`BaggingClassifier` or `BaggingRegressor`) to improve the robustness of the neural network.
   - [ ] Combine the predictions from the base models and feed them into the neural network.

3. **Create and Implement the Stacking Ensemble**
   - [ ] Use `sklearn`'s `StackingClassifier` and `StackingRegressor` for stacking the base models.
   - [ ] Ensure that the pipeline can handle multioutput (classification + regression) tasks using `MultiOutputClassifier` or `MultiOutputRegressor` for combined output.
   - [ ] Test the ensemble to ensure it works correctly with multioutput.

4. **Evaluate Model Performance**
   - [ ] Run cross-validation to assess the model's performance across the two output types.
   - [ ] Report performance metrics separately for the classification and regression tasks (e.g., accuracy for classification and MSE for regression).
   - [ ] Compare the stacking ensemble performance with the base models individually.

#### Example Code Structure

```python
from sklearn.ensemble import StackingClassifier, StackingRegressor, RandomForestClassifier, RandomForestRegressor
from sklearn.svm import SVC, SVR
from sklearn.linear_model import RidgeClassifier, Ridge
from sklearn.neural_network import MLPClassifier
from sklearn.ensemble import BaggingClassifier
from sklearn.multioutput import MultiOutputClassifier, MultiOutputRegressor
from sklearn.ensemble import GradientBoostingClassifier, GradientBoostingRegressor

# Base models
base_classifiers = [
    ('svc', SVC()),
    ('rf_clf', RandomForestClassifier()),
    ('gboost_clf', GradientBoostingClassifier()),
    ('ridge_clf', RidgeClassifier())
]

base_regressors = [
    ('svr', SVR()),
    ('rf_reg', RandomForestRegressor()),
    ('gboost_reg', GradientBoostingRegressor()),
    ('ridge_reg', Ridge())
]

# Meta model (bagging neural network)
meta_model = BaggingClassifier(MLPClassifier(), n_estimators=10)

# Stacking classifiers
stacking_clf = StackingClassifier(estimators=base_classifiers, final_estimator=meta_model)
stacking_reg = StackingRegressor(estimators=base_regressors, final_estimator=meta_model)

# Multioutput models
multi_clf = MultiOutputClassifier(stacking_clf)
multi_reg = MultiOutputRegressor(stacking_reg)

# Fit the multioutput ensemble
multi_clf.fit(X_train, y_train_class)
multi_reg.fit(X_train, y_train_reg)

# Evaluate model performance
clf_scores = cross_val_score(multi_clf, X_test, y_test_class, scoring='accuracy')
reg_scores = cross_val_score(multi_reg, X_test, y_test_reg, scoring='neg_mean_squared_error')

print(f'Classification accuracy: {clf_scores.mean()}')
print(f'Regression MSE: {-reg_scores.mean()}')

**Comment by project-delphi** [comment]
reset to clean modules with tested code.
